File: Siscon_API_Backend/Siscon_API_Backend/Models/MoradorApartamento.cs
namespace Siscon_API_Backend.Models
{
    public class MoradorApartamento
    {
        public long Id { get; set; }
        public long? MoradorId { get; set; }
        public long? ApartamentoId { get; set; }

        public Morador Morador { get; set; }
        public Apartamento Apartamento { get; set; }
    }
}

File: FRONTEND/SisconFrontEnd/SisconFrontEnd/src/app/models/MoradorApartamento.ts
import { Apartamento } from "./Apartamento";
import { Morador } from "./Morador";

export class MoradorApartamento {
  public Id: number;
  public MoradorId: number | null;
  public ApartamentoId: number | null;

  public Morador: Morador;
  public Apartamento: Apartamento;

}

export class MoradorApartamentoPayload {
  data: MoradorApartamento[];
}

export class MoradorApartamentoPayloadUniqueResult {
  data: MoradorApartamento;
}

export class MoradorApartamentoFiltro {
  nomeMorador?: string;
  numeroApartamento?: string;
  nomePredio?: string;
}
